eox_nelp/third_party_auth/tests/test_pipeline.py

```diff
@@ -50,37 +63,66 @@ def test_user_not_associated(self):
 
-        self.assertEqual(None, pipe_output)
-        self.backend.strategy.storage.user.get_user.assert_called_with(username=test_uid)
+        self.assertIsNone(pipe_output)
+        self.backend.strategy.storage.user.get_user.assert_called_with(**{self.user_query: test_uid})
 
-    @data(
-        {"is_staff": True, "username": "1222333444"},
-        {"is_superuser": True, "username": "1222333555"},
-    )
-    def test_staff_user_return_forbidden(self, user_kwargs):
+    def test_multiple_user_match(self):
+        """Test the pipeline method try to match with uid but there are multiple matches.
+        Expected behavior:
+            - Expected logs of multiple returns
+            - Pipe method returns None.
+        """
+        test_uid = "1888222999"
+
+        self.backend.get_idp.return_value.get_user_permanent_id.return_value = test_uid
+        exc_msg = "return 7"
+        self.backend.strategy.storage.user.get_user.side_effect = MultipleObjectsReturned(exc_msg)
+        expected_log = [
+            f"INFO:{utils.__name__}:"
+            f"Pipeline tries to match user with uid({test_uid}) using {self.user_query}, "
+            f"but Multiple users found: {exc_msg}"
+        ]
+        with self.assertLogs(utils.__name__, level="INFO") as logs:
+            pipe_output = self.uid_pipe(self.request, self.backend, self.details, self.response)
+            self.assertIsNone(pipe_output)
+        self.assertListEqual(logs.output, expected_log)
+
+
+class SaferAssociaciateUserByNationalIdTestCase(SaferAssociateUserUsingUid, TestCase):
+    """Test case for `safer_associate_user_by_national_id` method"""
+    uid_pipe = staticmethod(safer_associate_user_by_national_id)
+    user_query = "extrainfo__national_id"
+
+    def test_user_associate_uid_with_national_id(self):
         """Test the pipeline method try to match with uid, the user with matched username exists
-        but is staff or superuser.
+        and is returned.
         Expected behavior:
-            - The pipe method returns an HttpResponseForbidden object.
-            - The pipe method response has a 403 forbidden status.
             - Strategy storage get_user method is called with desired params.
+            - The method returns the desirect with `user` and `is_new` keys.
         """
-        test_uid = user_kwargs["username"]
-        past_user, _ = User.objects.get_or_create(**user_kwargs)
+        test_uid = "1777888999"
+        past_user, _ = User.objects.get_or_create(
+            username=test_uid,
+        )
 
         self.backend.get_idp.return_value.get_user_permanent_id.return_value = test_uid
         self.backend.strategy.storage.user.get_user.return_value = past_user
 
-        pipe_output = safer_associate_username_by_uid(self.request, self.backend, self.details, self.response)
+        pipe_output = safer_associate_user_by_national_id(self.request, self.backend, self.details, self.response)
```

Comment:
```suggestion
        pipe_output = self.uid_pipe(self.request, self.backend, self.details, self.response)
```
???  I Think it's the same but this suggestion keeps the style of other tests